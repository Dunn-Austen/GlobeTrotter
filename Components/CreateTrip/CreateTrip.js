import React, { useState } from 'react';
import { StyleSheet, Text, View, TextInput, TouchableOpacity, ScrollView } from 'react-native';
import { createNewTrip } from '../../apiCalls';

export const CreateTrip = ({ navigation, route }) => {
  const { handleTripsFetch } = route.params;

  let [origin, setOrigin] = useState('');
  let [name, setName] = useState('');

  const handleClick = () => {
    createNewTrip(name, origin)
      .then(returnedTripData => {
        handleTripsFetch();
        navigation.navigate('Add Destinations', {tripId: returnedTripData.data.createTrip.trip.id, handleTripsFetch})
      })
      .catch(error => {
        console.log(error)
      });
  };

  return (
    <ScrollView
      contentContainerStyle={styles.container}
      keyboardShouldPersistTaps='handled'
    >
      <Text style={styles.label}>Name</Text>
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.input}
          value={name}
          placeholder='Trip Name'
          onChangeText={(text) => setName(text)}
        />
      </View>
      <Text style={styles.label}>Starting Location</Text>
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.input}
          value={origin}
          placeholder='City, Country'
          onChangeText={(text) => setOrigin(text)}
        />
      </View>
      <TouchableOpacity
        activeOpacity={.8}
        style={styles.addDestination}
        onPress={() => handleClick()}
      >
        <Text style={{ color: '#0D47A1', fontSize: 20, fontWeight: 'bold' }}>Add A Destination</Text>
      </TouchableOpacity>
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#1E88E5',
    height: '100%',
    alignItems: 'center',
    padding: 15
  },
  inputContainer: {
    width: '100%',
    alignItems: 'center',
    backgroundColor: '#fff',
    marginBottom: 20,
    marginTop: 5,
    borderRadius: 5,
    height: 70
  },
  label: {
    color: '#fff',
    fontSize: 20
  },
  input: {
    justifyContent: 'center',
    height: '100%',
    width: '100%',
    fontSize: 20,
    paddingHorizontal: 20
  },
  addDestination: {
    backgroundColor: '#2CCB70',
    width: '100%',
    height: 50,
    borderRadius: 5,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 15
  },
  submitDestination: {
    backgroundColor: '#FDD835',
    width: '100%',
    height: 50,
    borderRadius: 5,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 15
  }
})


export default CreateTrip;