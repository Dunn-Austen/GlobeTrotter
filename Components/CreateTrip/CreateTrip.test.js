import React from 'react';
import CreateTrip from './CreateTrip';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<CreateTrip />).toJSON();
  expect(tree).toMatchSnapshot();
});