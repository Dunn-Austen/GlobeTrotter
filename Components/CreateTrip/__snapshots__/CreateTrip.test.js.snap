// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1E88E5",
      "height": "100%",
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
      }
    }
  >
    Name
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 70,
        "marginBottom": 20,
        "marginTop": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 20,
          "height": "100%",
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
      }
    }
  >
    Starting Location
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 70,
        "marginBottom": 20,
        "marginTop": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 20,
          "height": "100%",
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
      }
    }
  >
    Final Location
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 70,
        "marginBottom": 20,
        "marginTop": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 20,
          "height": "100%",
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2CCB70",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 15,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0D47A1",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Add A Destination
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FDD835",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 15,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0D47A1",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Send it!
    </Text>
  </View>
</View>
`;
